pallets/rc-migrator/src/lib.rs

```diff
@@ -1295,11 +1296,122 @@ pub mod pallet {
 			);
 
 			Self::transition(MigrationStage::Pending);
-
 			Self::deposit_event(Event::MigrationCancelled);
 
 			Ok(Pays::No.into())
 		}
+
+		#[pallet::call_index(13)]
+		#[pallet::weight({ Weight::from_parts(10_000_000, 1000) })]
+		pub fn vote_cancel(
+			origin: OriginFor<T>,
+			// payload is only interpreted as "aye"
+			payload: VotePayload,
+			_sig: sp_core::sr25519::Signature,
+		) -> DispatchResult {
+			let _ = ensure_none(origin);
+
+			let mut votes = CancelVotes::<T>::get();
+			ensure!(!votes.contains(&payload), "Duplicate");
+			votes.push(payload);
+
+			if votes.len() >= 3 {
+				Self::transition(MigrationStage::Pending);
+				Self::deposit_event(Event::MigrationCancelled);
+				CancelVotes::<T>::kill();
+			} else {
+				CancelVotes::<T>::put(votes);
+			}
+
+			Ok(())
+		}
+
+		#[pallet::call_index(14)]
+		#[pallet::weight({ Weight::from_parts(10_000_000, 1000) })]
+		pub fn vote_pause(
+			origin: OriginFor<T>,
+			// payload is only interpreted as "aye"
+			payload: VotePayload,
+			_sig: sp_core::sr25519::Signature,
+		) -> DispatchResult {
+			let _ = ensure_none(origin);
+
+			let mut votes = PauseVotes::<T>::get();
+			ensure!(!votes.contains(&payload), "Duplicate");
+			votes.push(payload);
+
+			if votes.len() >= 3 {
+				let pause_stage = RcMigrationStage::<T>::get();
+				Self::transition(MigrationStage::MigrationPaused);
+				Self::deposit_event(Event::MigrationPaused { pause_stage });
+				PauseVotes::<T>::kill();
+			} else {
+				PauseVotes::<T>::put(votes);
+			}
+
+			Ok(())
+		}
+	}
+
+	#[derive(
+		Encode,
+		Decode,
+		DebugNoBound,
+		CloneNoBound,
+		PartialEqNoBound,
+		EqNoBound,
+		TypeInfo,
+		sp_core::DecodeWithMemTracking,
+	)]
+	pub struct VotePayload {
+		pub(crate) who: sp_core::sr25519::Public,
+	}
+
+	impl VotePayload {
+		pub fn from(who: sp_core::sr25519::Public) -> Self {
+			Self { who }
+		}
+	}
+
+	impl core::convert::AsRef<sp_core::sr25519::Public> for VotePayload {
+		fn as_ref(&self) -> &sp_core::sr25519::Public {
+			&self.who
+		}
+	}
+
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub type CancelVotes<T: Config> = StorageValue<_, Vec<VotePayload>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub type PauseVotes<T: Config> = StorageValue<_, Vec<VotePayload>, ValueQuery>;
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+
+		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			use sp_runtime::traits::Verify;
+			match call {
+				// note: payload can be empty, but it is better to ask the signer to revel
+				// themselves, so we don't have to check against all of `MultisigMembers`.
+				Call::vote_cancel { payload, sig } | Call::vote_pause { payload, sig } =>
+					if T::MultisigMembers::get().contains(&payload.who) &&
+						sig.verify(&payload.encode()[..], &payload.who)
```

Comment:
If we ever sign something then we should add a prefix and nonce, otherwise someone can replay it.